src/basic/btrfs-util.c

```diff
@@ -917,6 +917,10 @@ int btrfs_resize_loopback_fd(int fd, uint64_t new_size, bool grow_only) {
         if (new_size < 16*1024*1024)
                 new_size = 16*1024*1024;
 
+        /* Linux VFS cannot handle file sizes > 8EiB, because off_t is signed 64bit */
+        if (new_size > INT64_MAX)
+                new_size = INT64_MAX;
+
```

Comment:
I have doubts (and I don't care:-) about >8EiB use-cases, but "set-limit infinity" is valid and documented use-case.